src/env-inl.h

```diff
@@ -129,10 +129,19 @@ inline v8::Local<v8::String> Environment::AsyncHooks::provider_string(int idx) {
   return providers_[idx].Get(isolate_);
 }
 
+inline void Environment::AsyncHooks::enable_sanity_check() {
+  // fields_ does not have the += operator defined
+  fields_[kSanityCheck] = fields_[kSanityCheck] + 1;
+}
+
 inline void Environment::AsyncHooks::push_ids(double async_id,
                                               double trigger_id) {
-  CHECK_GE(async_id, -1);
-  CHECK_GE(trigger_id, -1);
+  // Since async_hooks is experimental, do only perform the sanity check
+  // when async_hooks is enabled.
+  if (fields_[kSanityCheck] > 0) {
+    CHECK_GE(async_id, 0);
+    CHECK_GE(trigger_id, 0);
+  }
```

Comment:
hm. this is cool for release builds, but i think they should also always be on in debug builds.

Reply:
They are always on in tests, and you can `alias node='node --force-async-hooks-check' if you like. I think it is important that we have them opt-in in release builds too.

Reply:
And `export NODE_OPTIONS=--force-async-hooks-check`.